### ISSUE TITLE
Displaying archived/unarchived conversations appropriately, un-archiving conversations

### ISSUE THREAD

**Issue by nikilselvam** [psybuzz/goldenconversations#47]
Now that conversations can be archived, should we start displaying conversations in their appropriate sections (homepage vs. Hall of Fame) depending on if they're archived or not? We could also wait until Tony is done designing the Hall of Fame section before displaying the conversations.

Also, should we think about implementing functionality to allow people to un-archive conversations? This way users can move conversations both ways rather than just one way.

**Comment by psybuzz** [comment]
I think it would be a good idea to have the final design before implementing the Hall of Fame itself, but it seems like we could at least start a PR that simply shows the HoF convos unstyled for now, and then we add in the styling later.

Ah!  I didn't even think about un-archiving, but that's a great idea Nikil!  The functionality seems like it would be the exact same logic, except setting values to "false" instead of "true", I imagine.  I can add un-archiving as another Trello task.

**Comment by tonyxj** [comment]
@psybuzz @nikilselvam Guys, my thought was that we should keep all conversations on the homepage, sorted by the ones that are lasted active. Even with our usage, it didn't feel necessary to remove conversations for ephemerality or for saving to HOF. 
In terms of the front-end, just surface three sets of things:

Conversations details: Title, People involved & number favourited (# of people archiving)
Stats [stretch goal]: It'd be great if we can add in some stats as I mocked up: Total convos, gold plated convos, golden convos
Remove from HOF link

I can style up the pages once the html is there.

**Comment by nikilselvam** [comment]
@tonyxj Ah, gotcha. I see where you're coming from with regards to having conversations on both the home page and the HOF, but I also feel like this structure gives a sense of duplicity. I think that there's value to having each conversation in only one area (either the home page or the Hall of Fame). That being said, having conversations only in the HOF could discourage users from contributing to them any further since they are no longer on the home page. What do you think @psybuzz ?

Also, I realized one thing that the archived functionality needs to do is to change the option on the side of each conversation from `Add to Hall of Fame` to `Remove from Hall of Fame`. I think that the solution here is to use Handlebars in conversation.html and to check whether or not the user has already archived the conversation in order to update this text. I'm not sure how to go about getting this information from the backend, however. @psybuzz Could you help me with getting this information to conversation.html so that we can update the `Add/remove to Hall of Fame` option properly?

**Comment by psybuzz** [comment]
tldr at the bottom!

@nikilselvam Sure!  In order to reflect the Hall of Fame on the frontend, the use of Handlebars, as you suggested, is a great way to go about it.  In home.js, around lines 88-110, there is a Handlebars template defined that takes data from the backend.  I'd suggest that we use that as a starting point to work from.  Around those lines, we might be able to simply insert `{{hallOfFame}}` with some Handlebars logic to conditionally add a hall-of-fame CSS class.

On a related note, I started looking at our client-side JS code, and I'm finding less and less of it makes sense.  I think that we need an organizational overhaul to sort some of the logic, as it's getting harder and harder to find things, for me at least.

@tonyxj @nikilselvam On the question of how to display "famed" conversations, I think it could go either way.  If we really reduce this behavior, I believe it comes down to what we expect.  Do you "favorite/archive" a conversation while it is still active?  If so, then having HoF convos on the same page makes sense for users that naturally open them often (essentially like adding/removing a star).  If not, then having them on the same page could just add clutter, and it would be better to have them appear only when the user explicitly wants to see JUST the golden convos.

tldr: Rendering w/ Handlebars logic sounds good, let's start around lines 88-110 in home.js.  Also, we need to refactor/organize a lot of our client-side code.  I think the golden convo display behavior would work either way.  I think we should choose one and change it in a later iteration if needed.

**Comment by tonyxj** [comment]
**Regarding Refactor Client Code**
Yes. I'm for it. I think back when I implement a lot of the front end JS, there was a lot of trial and error to make things work. It could definitely use some cleaning up, not to mention documentation!

**Regarding HOF Duplicity** 
I think we should keep all conversations active in the app. It'd be unwelcoming to have golden conversations be frozen, or even give people that sense. I think the great way to celebrate the fact that everyone likes the convo so much to add it to the HOF is to in fact keep the participants talking in it, rather than HOF's connotation of 'retirement'. Removing it from the flow of convos on the homepage might trigger that "this is something that's been done" / or the sense of the convo is now retired (kinda like an athelete), which is why I suggested keep it at both place. By doing so, the HOF is a place for you, user to see which ones are golden, appreciate the fact that other people also cherish those convos. So I think we should perhaps keep it there. Also, I'll be tweaking some of the copy, especially for where we show 'archiving', since this is mostly an internal lingo, I'll be changing them to align to our 'add to TOC'. Also I think archiving suggest an idea more aligned to deletion that doesn't really reflect the nature making it gold.

Because of the HOF connotation or retirement, are there other copy we can try? Perhaps say 'make it gold?' or simply something like 'Like this conversation? Save it!' 
What do you guys think? @nikilselvam @psybuzz

**Comment by nikilselvam** [comment]
Sorry for the delay in response on this one, guys! It totally slipped my mind.

@psybuzz Thanks for your help in getting Handlebars going! I think that makes sense, and I'll look into implementing the un-archive functionality.

With regards to HOF conversations being active, I think this makes sense the more and more I think about it. I do agree that 'Hall of Fame' connotes retirement, and we certainly don't want to discourage users from having conversations. I like the idea of changing our copy to represent the idea of golden conversations more than Hall of Fame as we move forward. What do you guys think? @psybuzz @tonyxj

**Comment by tonyxj** [comment]
I think that'll be a good move, keeping it golden!
